### PR TITLE
Add test-only standalone signer creation

### PR DESCRIPTION
**Pull request by alnoki**
Adds test-only standalone signer creation, independent of an Aptos-style account.

@davidiw @wrwg @alinush @chen-robert

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/5433)

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/account.move [hunk 1]
```diff
@@ -129,6 +129,10 @@ module aptos_framework::account {
 
     native fun create_signer(addr: address): signer;
 
+    #[test_only]
+    /// Create signer for testing, independently of an Aptos-style `Account`.
+    public fun create_signer_for_test(addr: address): signer {create_signer(addr)}
```
Comment (alinush) [inline aptos-move/framework/aptos-framework/sources/account.move:134]: Add spaces after`{` and before `}`.

**Comment by alinush** [comment]
Thank you! Straightforward PR. I recall I wanted something like this at some point.

Two thoughts:

 - might we want this propagated further down in `aptos_stdlib` rather than `aptos_framework`?
 - @wrwg, @davidiw are we comfortable exporting this? What if there's ever a bug in test-only visibility?

File: aptos-move/framework/aptos-framework/sources/account.move (hunk 1 above)
Comment (alnoki) [reply aptos-move/framework/aptos-framework/sources/account.move]: @alinush 7612b0740ceb9ff1534c3ab4ce0d881f87b87270 adds space padding per your request.

**Comment by alnoki** [comment]
> * might we want this propagated further down in `aptos_stdlib` rather than `aptos_framework`?

I included this in `account.move` because that is where the native function `create_signer` is located. If this is propagated to `aptos_stdlib` I would advise moving the native function there as well - except that would break the ABI in the deployed mainnet version if I am not mistaken. Hence I suggest keeping it here to keep the two next to one another.

**Comment by alnoki** [comment]
> * might we want this propagated further down in `aptos_stdlib` rather than `aptos_framework`?

I included this in `account.move` because that is where the native function `create_signer` is located. If this is propagated to `aptos_stdlib` I would advise moving the native function there as well - except that would break the ABI in the deployed mainnet version if I am not mistaken. Hence I suggest keeping it here to keep the two next to one another.

**Review by davidiw — APPROVED** [review]
I'm fine with this.

We could theoretically make create_signer in a lot of places. I did that for genesis.move :)
I don't think move-std or aptos-std should know about signer or accounts

**Review by alinush — APPROVED** [review]

**Pull request merged**